front-end/src/domains/game-board/components/game-board/game-hex.component.tsx

```diff
@@ -0,0 +1,95 @@
+import React, {useState} from "react";
+
+// type Node = {
+//   type: NodeType;
+// }
+
+type NodeType = "water" | "space" | "concrete" | "fuel" | "food" | "player" | "metal" | "empty";
+// const landTypes: NodeType[] = ["water", "concrete", "fuel", "food", "metal"];
+
+function typeToColor(type: NodeType): string {
+  switch (type) {
+    case "water":
+      return "#4040ff"; // Blue
```

Comment:
This is another great example of where an Enum can come to the rescue! We can tie certain strings to certain values and make that its own type that TypeScript can enforce:

```suggestion
enum ColorType {
  WATER = "#4040ff", // Blue
  SPACE = "#000000", // Black
  CONCRETE = "#808080", // Grey
  FUEL = "#8d8b43", // Yellow
}
```
So on and so forth 😎

Reply:
Maybe migrate these enums into a top-level `constants.ts` file. This will keep them extensible and reusable throughout the app. Then, there's no need to copy/paste a bunch of easily-mistaken strings all over the app 😄.